packages/devtools_app/lib/src/shared/ui/hover.dart

```diff
@@ -66,6 +66,20 @@ String wordForHover(double dx, TextSpan line) {
   return word;
 }
 
+bool isPrimitiveValueOrNull(String valueAsString) {
+  final isNull = valueAsString == 'null';
+  final isBool = valueAsString == 'true' || valueAsString == 'false';
+  final isInt = int.tryParse(valueAsString) != null;
+  final isDouble = double.tryParse(valueAsString) != null;
+  final firstChar = valueAsString[0];
+  final lastChar = valueAsString[valueAsString.length - 1];
```

Comment:
can we guarantee the string is non-empty at this point? should we add an assert?

Reply:
verified that it's an appropriate length to do a string check